scalableminds/webknossos
# Simple fix to speed up dataset gallery

This is a very simple proposal to speed up the dataset gallery page for instances with a single organization. Previously in that case, all datasets (and their thumbnails) were shown and we thought about implementing pagination and/or viewport dependent rendering.
Now, when loading the page only the first 6 datasets are shown. A "Show more" link can be clicked to show all datasets (same as in the multiple-organizations-case, as on demo.webknossos).
I would argue that this alone will result in a decrease in server load as in most cases not all thumbnails are loaded. The search box still searches through all datasets and works as before.

If users complain about having to click that button or are doing it very often, we can always implement viewport dependent rendering later, I'd say :)

/cc @fm3 

### URL of deployed dev instance (used for testing):
- https://speedupspotlight.webknossos.xyz

### Steps to test:
- Open the gallery dataset view. Clicking "Show More" should show all datasets. Even having not clicked on "Show More", all datasets should be searchable.

### Issues:
- fixes #3442

------
